From 8b3e20bb36dde4bf4309503e99e2e3f286cfcdd3 Mon Sep 17 00:00:00 2001
From: Nicholas Bishop <nicholasbishop@google.com>
Date: Wed, 8 Mar 2023 16:16:29 -0500
Subject: [PATCH] make: add option to disable building/installing the docs

`ENABLE_DOCS=0 make` will remove the docs subdirectory from the list of
targets. This is useful if the target OS does not ship manpages.

Signed-off-by: Nicholas Bishop <nicholasbishop@google.com>
---
 Makefile                | 5 ++++-
 src/include/defaults.mk | 4 ++++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index c896fc38..5f1dc886 100644
--- a/Makefile
+++ b/Makefile
@@ -7,7 +7,10 @@ include $(TOPDIR)/src/include/defaults.mk
 include $(TOPDIR)/src/include/coverity.mk
 include $(TOPDIR)/src/include/scan-build.mk
 
-SUBDIRS := src docs
+SUBDIRS := src
+ifeq ($(ENABLE_DOCS), 1)
+SUBDIRS += docs
+endif
 
 all : | efivar.spec src/include/version.mk prep
 all clean install prep :
diff --git a/src/include/defaults.mk b/src/include/defaults.mk
index 630da237..4da0cde3 100644
--- a/src/include/defaults.mk
+++ b/src/include/defaults.mk
@@ -122,4 +122,8 @@ COMMIT_ID=$(shell git log -1 --pretty=%H 2>/dev/null || echo master)
 
 NAME=efivar
 
+# Docs are enabled by default. Set ENABLE_DOCS=0 to disable
+# building/installing docs.
+ENABLE_DOCS ?= 1
+
 # vim:ft=make
